libs/feature/teaching/src/lib/course/course-learnpath-editor/skill-lesson-linker.tsx

```diff
@@ -0,0 +1,163 @@
+import { Mesh, SkillFormModel } from "@self-learning/types";
+import { SelectSkillDialog } from "../../skills/skill-dialog/select-skill-dialog";
+import { IconButton } from "@self-learning/ui/common";
+import { PlusIcon } from "@heroicons/react/24/solid";
+import { LessonSelector, LessonSummary } from "../course-content-editor/dialogs/lesson-selector";
+import { useState } from "react";
+
+export function SkillLessonLinker({
+	repositoryId,
+	onClick: submitMesh,
+	changeDisplayedMesh: changeCurrentMesh,
+	displayedMesh: displayedMesh
+}: {
+	repositoryId: string;
+	onClick: (mesh: Mesh) => void;
+	changeDisplayedMesh: (mesh: Mesh) => void;
+	displayedMesh: Mesh;
+}) {
+	const [requiredSkillSelectorOpen, setRequiredSkillSelectorOpen] = useState(false);
+	const [gainedSkillSelectorOpen, setGainedSkillSelectorOpen] = useState(false);
+	const [lessonSelectorOpen, setLessonSelectorOpen] = useState(false);
+
+	function handleSubmit() {
+		const mesh: Mesh = {
+			requiredSkills: displayedMesh.requiredSkills,
+			lesson: displayedMesh.lesson,
+			gainedSkills: displayedMesh.gainedSkills
+		};
+		submitMesh(mesh);
+	}
```

Comment:
das verstehe ich nicht ganz. displayed mesh wird einfach direkt übergeben. 

es wirkt so, als würde hier seperation of concerns nicht gut sein. 

eine obere komponente kennet displayedMesh und die submitMesh methode. Eine untere komponente bringt die beiden dann zusammen, ohne zu wissen warum. 

wirkt eher so als wenn die onClick methode eher so aussehen sollte: 

```ts
<SkillLessonLinker onClick={ () => addMesh(displayedMesh) }
```